xbmc/pvr/windows/GUIViewStatePVR.cpp

```diff
@@ -20,9 +20,10 @@ using namespace PVR;
 
 CGUIViewStateWindowPVRChannels::CGUIViewStateWindowPVRChannels(const int windowId, const CFileItemList& items) : CGUIViewStatePVR(windowId, items)
 {
-  AddSortMethod(SortByChannelNumber, 549, LABEL_MASKS("%L", "", "%L", ""));      // "Number"      : Filename, empty | Foldername, empty
-  AddSortMethod(SortByChannel,       551, LABEL_MASKS("%L", "", "%L", ""));      // "Name"        : Filename, empty | Foldername, empty
-  AddSortMethod(SortByLastPlayed,    568, LABEL_MASKS("%L", "%p", "%L", "%p"));  // "Last played" : Filename, LastPlayed | Foldername, LastPlayed
+  AddSortMethod(SortByClientChannelOrder,  19315, LABEL_MASKS("%L", "", "%L", ""));      // "Backend number" : Filename, empty | Foldername, empty
```

Comment:
This sort method should only be available if "use backend sort order" is activated in settings.

I would add this sort method at the end, not at the start - given that this is  a very special sort method, compared to the other ones.

Reply:
It being at the end of the list makes sense but it's not dependent on "use backend sort order". You may like the Kodi numbering but still want the backend order.

Besides, "use backend sort order" is sorting by the backend numbers so it relates to something different. The text for "use backend sort order" could be changed to make it's meaning more clear.

Reply:
> It being at the end of the list makes sense but it's not dependent on "use backend sort order". You may like the Kodi numbering but still want the backend order.

 Good point.

> Besides, "use backend sort order" is sorting by the backend numbers 

 ... because currently we do not have any other backend order than the numbers. With your PR this changes and thus renaming the setting label could make sense, yes.

Reply:
BTW: What's the behavior if I sort by backend order and backend supplies iPosition == 0 for all channels?

Reply:
Actually, I just checked and "use backend order" appears to have no effect whatsoever. I guess the left nav sorting was added after this.

Reply:
can of worms?

Reply:
Maybe should remove "Use backend order" from settings.

Reply:
Many worms...

Reply:
The more I think about this, the more I ask myself whether the feature introduced by this PR makes sense. ;-)

Why is it not enough to use the backend channel numbers as sort criteria? Why do we need something "overloading" the numbers. What's the benefit/use case in assigning a channel the number "1" but not wanting to see it at the beginning of channel list, but somewhere in the middle or elsewhere? Isn't the (sole) purpose of channel numbers to order the channels to match users preference? So confusing...

Reply:
If order is zero for all channels that sort does nothing.

Reply:
Have you seen my question re general sense of this PR?

Reply:
I have my difficulties in seeing the use case for this feature.

Reply:
I think it does make sense. My wife has her favourites group on our PVR, morning channels at the top, evening channels at the bottom.

Reply:
You can also order channels by name and last watched.

Reply:
Fair enough.

Reply:
> Maybe should remove "Use backend order" from settings.

 Why?

Reply:
Because the setting doesn't appear to have any effect. PVRChannelGroup sorts the channels, but then they are sorted again by the left nav sort. Or am I missing something...

Reply:
> Because the setting doesn't appear to have any effect.

 The question is: should we fix that?

Reply:
But as you said, can of worms ;) Maybe best to not touch until a rewrite.

Reply:
Wait: The purpose of the setting is not to define the order of the channels on screen windows. Its the purpose to respect the backend sort order (defined be the backend channel numbers) when creating local channel numbers!

Backend:
channel A, number 1
channel B, number 5
channel C, number 9

Kodi: use backend sort order active, use backend channel numbers not active
channel A, number 1
channel B, number 2
channel C, number 3

Activating use channel backend numbers implies use backend channel order. The latter gets changed by your PR.

Reply:
I see. That makes sense.

This PR only changes the order on the screen though. It's doesn't effect the sorting within PVRChannelGroup so should not change the latter.

Reply:
To conclude:

* your feature makes sense
* settings "use backend channel order" makes sense

;-)

Reply:
I just spotted something, if you set a different channel number per group for the same channel it has no effect. That’s ones for a different PR for different day 😉

Reply:
I completely agree with your summary!!!